backends/nxp/runtime/NeutronDriver.h

```diff
@@ -0,0 +1,252 @@
+
+  /// Neutron kernels buffer address.
+  /// The Neutron kernels are generated by the Neutron converter tool.
+  /// The kernels buffer, 16 bytes aligned, is allocated and initialized by the
+  /// application or ML framework. The kernels buffer address is passed by
+  /// reference to the Neutron-firmware. The kernels buffer is specific for a
+  /// given ML model.
+  const void* kernels;
+
+  /// Timeout seconds for the microcode running.
+  /// This timeout is the uplimit seconds that a user expect to complete,
+  /// default 60.
+  uint32_t timeoutSeconds;
+
+} NeutronModelConfig;
+
+typedef struct {
+  /// The input buffers of the model.
+  /// The input buffers are allocated and initialized by the application or ML
+  /// framework. The input buffers are passed by reference to the Neutron
+  /// firmware.
+  const void** inputs;
+
+  /// The output buffers of the model.
+  /// The output buffers are allocated by the application or ML framework.
+  /// The output buffers are passed by reference to the Neutron firmware.
+  void** outputs;
+
+  /// Scratch buffer required for computing model intermediate results.
+  /// If NULL, this buffer has to be allocated by the driver.
+  void* scratch;
+
+  /// Scratch buffer required for prefetching model weights from FLASH to SRAM.
+  /// This buffer is used only for Neutron-C targets when the weight prefetch
+  /// option was explicitly used. If NULL, this buffer has to be allocated by
+  /// the driver.
+  void* scratchWeights;
+
+} NeutronDataConfig;
+
+typedef struct {
+  /// Sets whether tracing should be executed during firmware run or not.
+  /// If set to 0, tracing will not run.
+  /// If set to 1 - kernel level tracing.
+  /// If set to 2 - job level tracing.
+  /// If set to 3 - mixed level tracing
+  uint32_t traceConfig;
+
+  /// Buffer to store collected trace data.
+  /// If it is NULLPTR, driver will allocate the memory, otherwise, application
+  /// can.
+  char* traceBuffer;
+
+  /// What is the allocated memory for buffer. Needed to check if appending
+  /// string will be out of bounds. Application should set this, if the buffer
+  /// is allocated by application, otherwise driver will set the value.
+  size_t traceBufferSize;
+} NeutronTraceConfig;
+
+/// This structure contains the prototypes for functions that have a custom
+/// implementation. Any new functions or variables must be added at the end.
+typedef struct {
+  /// This function performs the copying from FLASH to SRAM.
+  void (*copy)(void* dst, void* src, uint32_t size, uint32_t channel);
+  /// This is a blocking function that checks if the current copy has finished.
+  void (*wait)(uint32_t channel);
+} NeutronConfig;
+
+/* Invalid handle, returned by neutronModelPrepare() if an error occurred. */
+#define NEUTRON_INVALID_HANDLE NULL
+
+/// - Initialize the Neutron Driver library, setting initial values, do memory
+/// allocation
+///   for internal data structures, do memory mapping.
+NeutronError neutronInit();
```

Comment:
can these functions be in a `neutron` namespace? 
Where are we calling this specific function to init the lib?

Reply:
It is called in surrounding rtos code, much like ethosu_init